horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -58,32 +66,222 @@ public class RunServiceTest extends BaseServiceTest {
    @Inject
    EventBus eventBus;
 
+   private static final int POLL_DURATION_SECONDS = 10;
+
+   @org.junit.jupiter.api.Test
+   public void testStraightforwardTransformationEmptyResult(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      NamedJsonPath path = createExampleNamedJsonPath("foo", "$.value");
+      Schema schema = createExampleSchema(info, true);
+
+      assertNotNull(schema.id);
+      Transformer transformer = createExampleTransformerWithJsonPath("acme", schema, "", path);
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, transformer);
+      uploadRun("{\"corporation\":\"acme\"}", test.name);
+
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertTrue(node.isEmpty());
+   }
+
+   private NamedJsonPath createExampleNamedJsonPath(String name, String path) {
+      NamedJsonPath jsonpath = new NamedJsonPath();
+      jsonpath.name = name;
+      jsonpath.jsonpath = path;
+      return jsonpath;
+   }
+
    @org.junit.jupiter.api.Test
-   public void testDataSetCreated(TestInfo info) throws InterruptedException {
+   public void testTransformationWithoutSchema(TestInfo info) throws InterruptedException {
       BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
       Test test = createTest(createExampleTest(getTestName(info)));
-      Schema schema = createExampleSchema(info);
       setTestVariables(test, "Value", "value");
 
-      uploadRun(runWithValue(schema, 42).toString(), test.name);
+      dataSetQueue.drainTo(new ArrayList<>());
+      uploadRun( "{ \"foo\":\"bar\"}", test.name);
 
-      try (CloseMe ignored = roleManager.withRoles(em, Arrays.asList(TESTER_ROLES))) {
-         DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
-         assertNotNull(event);
-         assertNotNull(event.id);
-         Integer id = event.id;
-         DataSet ds = DataSet.findById(id);
-         assertNotNull(ds);
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertNotNull(node);
+      assertTrue(node.isArray());
+      assertTrue(node.isEmpty());
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutExtractors(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      Schema schema = createExampleSchema(info, false);
+
+      assertNotNull(schema.id);
+      NamedJsonPath path = null; // no extractors
+      Transformer transformer = createExampleTransformerWithJsonPath("acme", schema, "", path);
```

Comment:
Now thinking about it we need to test that if there are not extractors the function (which is empty in this case) is actually running.
The result of this test is actually a bit surprising to me - absence of the transformation function usually means that it outputs the same object that would be its input. However without any extractors the input is an empty object, and here it seems the output is the original object.

Reply:
In the absence of a meaningful transformer Function the original schema object is returned. Extractors present or not makes no difference.

Reply:
To double check here.
 An empty Function `""` should mean the JSON paths are kept. The Function is bypassed. Correct ?

Reply:
Yes.

Reply:
ok thanks.

Reply:
How about in the situation the Function is blank, no extractors, run data (with fields of data) does not contain a schema. What's expected ?